### DISCUSSION TITLE
queryClient.cancelQueries is triggering onError callback

### DISCUSSION THREAD

**Discussion by AdhamMoussa** [TanStack/query#3647] [Q&A]
I have a mutation with optimistic update
when the `queryClient.cancelQueries` run while the query is fetching, the onError callback of the Query is triggered
is this expected?
how to avoid that?

**Comment by TkDodo** [comment]
hmm, can you show that in a codesandbox reproduction?

**Reply by AdhamMoussa** [reply]
@TkDodo sorry I had this issue on v3.15.0
fixed by upgrading to v3.39.1
